packages/snap/tests/e2e/createVerifiableCredential.spec.ts

```diff
@@ -0,0 +1,224 @@
+  ['snap'],
+  // ['ceramic'],
+  // ['snap', 'ceramic'],
+];
+
+describe.each(methods)('Create VerifiableCredential with %s', (method) => {
+  let snapMock: SnapsGlobalObject & SnapMock;
+  let issuer: string;
+  let agent: Agent;
+
+  beforeAll(async () => {
+    snapMock = createMockSnap();
+    snapMock.rpcMocks.snap_manageState({
+      operation: 'update',
+      newState: getDefaultSnapState(account),
+    });
+    const ethereumMock = snapMock as unknown as MetaMaskInpageProvider;
+    agent = await getAgent(snapMock, ethereumMock);
+    global.snap = snapMock;
+    global.ethereum = snapMock as unknown as MetaMaskInpageProvider;
+
+    snapMock.rpcMocks.snap_dialog.mockReturnValue(true);
+  });
+
+  beforeEach(async () => {
+    await agent.clear({ options: { store: ['snap', 'ceramic'] } });
+  });
+
+  it('Should switch DID method', async () => {
+    const switchMethod = (await onRpcRequest({
+      origin: 'localhost',
+      request: {
+        id: 'test-id',
+        jsonrpc: '2.0',
+        method: 'switchDIDMethod',
+        params: {
+          didMethod: method,
+        },
+      },
+    })) as Result<string>;
+
+    if (isError(switchMethod)) {
+      throw new Error(switchMethod.error);
+    }
+
+    expect(switchMethod.data.substring(0, method.length)).toBe(method);
+    issuer = switchMethod.data;
+    expect.assertions(1);
+  });
+
+  describe.each(proofFormats)('Using Proof Format: %s', (proofFormat) => {
+    it('Should create a VerifiableCredential', async () => {
+      const vc = (await onRpcRequest({
+        origin: 'localhost',
+        request: {
+          id: 'test-id',
+          jsonrpc: '2.0',
+          method: 'createVC',
+          params: {
+            minimalUnsignedCredential: examplePayload,
+            proofFormat,
+          },
+        },
+      })) as Result<VerifiableCredential>;
+
+      if (isError(vc)) {
+        throw new Error(vc.error);
+      }
+
+      const validity = await agent.verifyCredential({ credential: vc.data });
+      expect(validity.verified).toBe(true);
+
+      if (typeof vc.data.issuer === 'string') {
+        expect(vc.data.issuer).toBe(issuer);
+      } else {
+        expect(vc.data.issuer.id).toBe(issuer);
+      }
+      expect(vc.data.proof.type).toBe(proofTypes[proofFormat]);
+      expect.assertions(3);
+    });
+
+    it.each(stores)(
+      'Should create and save a VerifiableCredential in %s ',
+      async (store) => {
+        const vc = (await onRpcRequest({
+          origin: 'localhost',
+          request: {
+            id: 'test-id',
+            jsonrpc: '2.0',
+            method: 'createVC',
+            params: {
+              minimalUnsignedCredential: examplePayload,
+              proofFormat,
+              options: {
+                save: true,
+                store,
+              },
+            },
+          },
+        })) as Result<VerifiableCredential>;
+
+        expect(isError(vc)).toBe(false);
```

Comment:
Probably don't the expect if we have the if statement 1 row lower.
The if statement is required.

Reply:
Also look in other places for the same thing.